core/templates/dev/head/pages/exploration_player/ExplorationEngineService.js

```diff
@@ -121,6 +123,22 @@ oppia.factory('ExplorationEngineService', [
         AlertsService.addWarning('Expression parsing error.');
         return;
       }
+      currentStateName = exploration.initStateName;
+      nextStateName = exploration.initStateName;
+
+      var interaction = exploration.getInteraction(exploration.initStateName);
+      var nextFocusLabel = FocusManagerService.generateFocusLabel();
+
+      var interactionId = interaction.id;
+      var interactionHtml = null;
+
+      if (interactionId) {
+        interactionHtml = ExplorationHtmlFormatterService.getInteractionHtml(
+          interactionId,
+          exploration.getInteractionCustomizationArgs(currentStateName),
+          true, nextFocusLabel);
+        FocusManagerService.setFocusIfOnDesktop(nextFocusLabel);
```

Comment:
This seems like it's in the wrong place. You probably want to store the focus label in the card object, and then call this in the conversation skin service once the card is visible. This is important for autofocusing correctly.

Reply:
You don't need to do it when browsing through history, since there's nothing to focus on.

Just when navigating to a new card, or when a new input/response pair appears, so that e.g. you autofocus to the text input field and the learner can just type their answer without having to click into the text input field.

Reply:
Yeah, done. setFocus() is already called at the right location when submitting answer. Added it to the place where initial card is loaded following a previous comment and removed it from here.